src/js/view/view.js

```diff
@@ -505,7 +510,12 @@ function View(_api, _model) {
             _api.pause({ reason: 'interaction' });
         });
 
-        controls.on('all', _this.trigger, _this);
+        controls.on('all', (type, event) => {
+            if (event && event.doNotForward) {
```

Comment:
Is this to prevent `controller` from firing `dismissFloating` event?
Outstream listens to the `dismissFloating` view event: https://github.com/jwplayer/jwplayer-commercial/blob/master/src/js/view/view-outstream.js#L46

Reply:
Commercial view and this are the same.  This prevents events from being forwarded through the public API. dismissFloating is not a documented event or a new event that we want to make public.

Reply:
But the commercial `view-outstream` is listening to `view.on('dismissFloating`)`, which is triggered in the line below with `_this.trigger`. Shouldn't the check be moved to `controller`?

Reply:
I was thinking commercial is also listening to controls (not the instance it's extending). I will look into this.